src/main/java/appeng/parts/networking/PartDenseCableCovered.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2014, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.parts.networking;
+
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+
+import appeng.api.networking.GridFlags;
+import appeng.api.networking.IGridHost;
+import appeng.api.networking.IGridNode;
+import appeng.api.networking.events.MENetworkChannelsChanged;
+import appeng.api.networking.events.MENetworkEventSubscribe;
+import appeng.api.networking.events.MENetworkPowerStatusChange;
+import appeng.api.parts.BusSupport;
+import appeng.api.parts.IPartCollisionHelper;
+import appeng.api.util.AECableType;
+import appeng.api.util.AEPartLocation;
+import appeng.helpers.Reflected;
+import appeng.util.Platform;
+
+
+public class PartDenseCableCovered extends PartCable
+{
+	@Reflected
+	public PartDenseCableCovered( final ItemStack is )
+	{
+		super( is );
+
+		this.getProxy().setFlags( GridFlags.DENSE_CAPACITY, GridFlags.PREFERRED );
+	}
+
+	@Override
+	public BusSupport supportsBuses()
+	{
+		return BusSupport.DENSE_CABLE;
+	}
+
+	@Override
+	public AECableType getCableConnectionType()
+	{
+		return AECableType.DENSE_COVERED;
+	}
+
+	@Override
+	public void getBoxes( final IPartCollisionHelper bch )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') The Cyclomatic Complexity of this method "getBoxes" is 21 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)